gnomad/assessment/summary_stats.py

```diff
@@ -355,6 +356,185 @@ def get_tx_expression_expr(
+    :param lof_csq_set: Set of LoF consequence strings. Default is None.
+    :param lof_label_set: Set of LoF consequence labels. Default is None.
+    :param lof_flag_set: Set of LoF consequence flags. Default is None.
+    :param lof_no_flags: Whether to filter to LoF variants with no flags. Default is
+        False.
+    :param lof_any_flags: Whether to filter to LoF variants with any flags. Default is
+        False.
+    :param lof_loftee_combinations: Whether to add combinations of LOFTEE and
+        consequence type filters. Default is False.
+    :param additional_csq_sets: Dictionary containing additional consequence sets.
+        Default is None.
+    :param additional_csqs: Set of additional consequences to keep. Default is None.
+    :param collapse_filters: Whether to collapse all filters into a single expression.
+        Default is False.
+    :return: BooleanExpression or Dict of BooleanExpressions for filtering consequences.
+    """
+    # Set up filters for specific consequences or sets of consequences.
+    csq_filters = {
+        **({"csq_set_lof": lof_csq_set or {}}),
+        **({f"csq_set_{l}": c for l, c in (additional_csq_sets or {}).items()}),
+        **({f"lof_csq_{c}": {c} for c in lof_csq_set or []}),
+        **({f"csq_{c}": {c} for c in additional_csqs or []}),
+    }
+
+    def _create_filter_by_csq(
+        t: Union[hl.Table, hl.MatrixTable],
+        csq_set: Union[Set, List, hl.expr.CollectionExpression],
+    ) -> hl.expr.BooleanExpression:
+        """
+        Create filtering expression for a set of consequences.
+
+        :param t: Input Table/MatrixTable.
+        :param csq_set: Set of consequences to filter.
+        :return: BooleanExpression for filtering by consequence.
+        """
+        if not isinstance(csq_set, hl.expr.CollectionExpression):
+            csq_set = hl.set(csq_set)
+
+        return csq_set.contains(t.most_severe_csq)
+
+    # Create filtering expressions for each consequence set.
+    ss_filter_expr = {
+        filter_name: _create_filter_by_csq(t, csq_set)
+        for filter_name, csq_set in csq_filters.items()
+    }
+
+    # Add filtering expressions for LoF consequence labels.
+    lof_labels = {
+        f"lof_{lof_label}": hl.or_else(t.lof == lof_label, False)
+        for lof_label in lof_label_set or []
+    }
+
+    # Add filtering expressions for LoF consequence flags.
+    lof_flags = {
+        f"lof_flag_{lof_flag}": hl.or_else(
+            t.lof_flags.split(",").contains(lof_flag), False
+        )
+        for lof_flag in lof_flag_set or []
+    }
+
+    # Add filtering expressions for HC LoF variants with no flags or any flags.
+    lof_hc_flags = {}
+    if lof_no_flags or lof_any_flags:
+        hc_expr = hl.or_else(t.lof == "HC", False)
+        if "no_lof_flags" in t.row:
+            no_lof_flags_expr = t.no_lof_flags
+        elif "lof_flags" in t.row:
+            no_lof_flags_expr = hl.is_missing(t.lof_flags) | (t.lof_flags == "")
```

Comment:
why would there be different behavior in the input 't' here? Just curious

Reply:
I could be wrong, but I think this is needed because of a version change. It used to only need to check for "", but now that doesn't work and it needs to check for missingness. This will allow it to work for either version